src/foundry/client/apps/sidebar/apps/world-config.d.mts

```diff
@@ -7,60 +7,54 @@ declare global {
      * @defaultValue `false`
      */
     create: boolean;
-
-    inWorld?: boolean | undefined;
   }
 
   /**
    * The World Management setup application
    * @typeParam Options - The type of the options object
    */
-  class WorldConfig<Options extends WorldConfigOptions = WorldConfigOptions> extends FormApplication<
-    Options,
-    Game.WorldData<foundry.packages.WorldData>
-  > {
+  class WorldConfig<Options extends WorldConfigOptions = WorldConfigOptions> extends FormApplication<Options, World> {
     /**
      * @defaultValue
      * ```typescript
      * foundry.utils.mergeObject(super.defaultOptions, {
      *   id: "world-config",
      *   template: "templates/sidebar/apps/world-config.html",
-     *   width: 600,
+     *   width: 620,
      *   height: "auto",
      *   create: false,
      * })
      * ```
      */
     static override get defaultOptions(): WorldConfigOptions;
 
-    static WORLD_KB_URL: "https://foundryvtt.com/article/game-worlds/";
+    /**
+     * A semantic alias for the World object which is being configured by this form.
+     */
+    get world(): World;
 
     override get title(): string;
 
     override activateListeners(html: JQuery): void;
 
+    // TODO: Implement GetDataReturnType
     override getData(options?: Partial<Options>): MaybePromise<object>;
 
-    /**
-     * @remarks This method returns `Promise<void>`.
-     */
-    protected override _onSubmit(event: Event): Promise<any>;
-
-    /**
-     * @remarks This method does not exist on WorldConfig and only exists to make the typescript compile!
-     */
-    protected _updateObject(...args: unknown[]): Promise<unknown>;
+    protected override _getSubmitData(updateData?: object | null): WorldConfig.FormData;
 
-    /**
-     * Update the world name placeholder when the title is changed.
-     * @internal
-     */
-    protected _onTitleChange(event: JQuery.TriggeredEvent): void;
+    protected override _updateObject(event: Event, formData: WorldConfig.FormData): Promise<number | void>;
 
     override activateEditor(
       name: string,
       options?: TextEditor.Options | undefined,
       initialContent?: string | undefined,
     ): ReturnType<FormApplication["activateEditor"]>;
   }
+
+  namespace WorldConfig {
+    type FormData = Record<string, boolean> & {
```

Comment:
Not quite correct - the FormData is the expected return for `world-config.hbs`, which basically amounts to `Partial<World["_source"]>`. If you want to be really precise you could enumerate the specific fields with a matching `name` property in the handlebars template. However, this is a low priority app because it's only available in the Setup screen, where systems & modules can't run code.

Reply:
So just `type FormData = Partial<World["_source"]>`? and if so, is it even worth putting in a new namespace vs just using that up above as the return type for `_getSubmitData` and the type of `formData`?

Reply:
Put it up there in case someone in the future decides to make it more particular, since technically not *every* property of World._source is available for configuration via the form (e.g. Version will never be included)